src/PhraseApp.php

```diff
@@ -0,0 +1,251 @@
+            'project_id' => $this->projectId,
+            'locale_id' => $this->getLocaleId($locale),
+        ]);
+
+        foreach ($translations as $translation) {
+            if ($translation['key']['name'] === "$domain::$key") {
+                return new Message($key, $domain, $locale, substr($translation['content'], strlen($domain) + 2));
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(Message $message)
+    {
+        try {
+            $response = $this->client->request('key.search', [
+                'project_id' => $this->projectId,
+                'locale_id' => $this->getLocaleId($message->getLocale()),
+                'q' => 'tags:' . $message->getDomain() . ' name:' . $message->getDomain() . '::' . $message->getKey(),
+            ]);
+
+            foreach ($response as $key) {
+                if ($key['name'] === $message->getDomain() . '::' . $message->getKey()) {
+                    $keyId = $key['id'];
+                    break;
+                }
+            }
+
+            if (! isset($keyId)) {
+                $response = $this->client->request('key.create', [
+                    'project_id' => $this->projectId,
+                    'locale_id' => $this->getLocaleId($message->getLocale()),
+                    'name' => $message->getDomain() . '::' . $message->getKey(),
+                    'tags' => $message->getDomain(),
+                ]);
+
+                $keyId = $response['id'];
+            }
+
+            $response = $this->client->request('translation.indexKeys', [
+                'project_id' => $this->projectId,
+                'key_id' => $keyId
+            ]);
+
+            if (empty($response)) {
+                $this->client->request('translation.create', [
+                    'project_id' => $this->projectId,
+                    'locale_id' => $this->getLocaleId($message->getLocale()),
+                    'key_id' => $keyId,
+                    'content' => $message->getDomain() . '::' . $message->getTranslation(),
+                ]);
+
+                return;
+            }
+
+            foreach ($response as $translation) {
+                if ($translation['locale']['name'] === $message->getLocale()) {
+                    $id = $translation['id'];
+                }
+                break;
+            }
+
+            if (! isset($id)) {
+                throw new StorageException('Translation id not found.');
+            }
+
+            $this->client->request('translation.update', [
+                'project_id' => $this->projectId,
+                'id' => $id,
+                'content' => $message->getDomain() . '::' . $message->getTranslation(),
+            ]);
+        } catch (\Throwable $e) {
+            throw new StorageException($e->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(Message $message)
+    {
+        $this->create($message);
```

Comment:
Be aware that the interface description is: "Update a translation. Creates a translation if there is none to update."